src/main/java/de/dennisguse/opentracks/TrackRecordActivity.java

```diff
@@ -0,0 +1,456 @@
+                setLockscreenPolicy();
+            }
+
+            if (PreferencesUtils.isKey(TrackRecordActivity.this, R.string.stats_keep_screen_on_while_recording_key, key)) {
+                setScreenOnPolicy();
+            }
+
+            if (key == null) return;
+
+            runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    TrackRecordActivity.this.invalidateOptionsMenu();
+                    trackController.update(true, recordingTrackPaused);
+                }
+            });
+        }
+    };
+
+    private MenuItem insertMarkerMenuItem;
+    private MenuItem markerListMenuItem;
+
+    private final OnClickListener recordListener = new OnClickListener() {
+        @Override
+        public void onClick(View v) {
+            if (recordingTrackPaused) {
+                // Paused -> Resume
+                updateMenuItems(false);
+                trackRecordingServiceConnection.resumeTrack();
+                trackController.update(true, false);
+            } else {
+                // Recording -> Paused
+                updateMenuItems(true);
+                trackRecordingServiceConnection.pauseTrack();
+                trackController.update(true, true);
+            }
+        }
+    };
+
+    private final OnClickListener stopListener = new OnClickListener() {
+        @Override
+        public void onClick(View v) {
+            trackRecordingServiceConnection.stopRecording(TrackRecordActivity.this, true);
+            Intent newIntent = IntentUtils.newIntent(TrackRecordActivity.this, TrackDetailActivity.class)
+                    .putExtra(TrackDetailActivity.EXTRA_TRACK_ID, trackId);
+            startActivity(newIntent);
+            updateMenuItems(true);
+            finish();
+        }
+    };
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        recordingTrackPaused = PreferencesUtils.isRecordingTrackPausedDefault(this);
+
+        contentProviderUtils = new ContentProviderUtils(this);
+        handleIntent(getIntent());
+
+        sharedPreferences = PreferencesUtils.getSharedPreferences(this);
+
+        trackRecordingServiceConnection = new TrackRecordingServiceConnection(bindChangedCallback);
+        trackDataHub = new TrackDataHub(this);
+
+        FragmentPagerAdapter adapter = new FragmentPagerAdapter(getSupportFragmentManager(), 1) {
+            @Override
+            public int getCount() {
+                return 3;
+            }
+
+            @NonNull
+            @Override
+            public Fragment getItem(int position) {
+                switch (position) {
+                    case 1:
+                        return ChartFragment.newInstance(false);
+                    case 2:
+                        return ChartFragment.newInstance(true);
+                    default: //0
```

Comment:
Why not make it 0?
And default to throw RuntimeException?

Reply:
Better.
I've changed TrackDetailActivity too.